File: src/test/java/com/github/qky666/selenidepom/test/java/mtp/pom/containerpom/ServicesRequiredErrorPage.java
package com.github.qky666.selenidepom.test.java.mtp.pom.containerpom;

import com.codeborne.selenide.SelenideElement;
import com.github.qky666.selenidepom.pom.Required;
import org.openqa.selenium.support.FindBy;

import java.time.Duration;

import static com.codeborne.selenide.Condition.text;

public class ServicesRequiredErrorPage extends MainFramePage {

    // Fields
    @FindBy(css = "div.servicios-principal") @Required public SelenideElement principal;
    @FindBy(css = "h1.h2") @Required public SelenideElement title;
    // badSelector generates an error in shouldLoadRequired
    @FindBy(css = "bad-selector") @Required public SelenideElement badSelector;
    @FindBy(css = "other-bad-selector") @Required public SelenideElement otherBadSelector;

    // Methods
    @Override
    public void customShouldLoadRequired(Duration timeout, String model, String lang) throws Throwable {
        super.customShouldLoadRequired(timeout, model, lang);
        title.shouldHave(text("Aseguramiento de la calidad"), timeout);
    }
}

File: src/test/kotlin/com/github/qky666/selenidepom/test/kotlin/mtp/pom/pages/services/ServicesRequiredErrorPage.kt
package com.github.qky666.selenidepom.test.kotlin.mtp.pom.pages.services

import com.codeborne.selenide.Selenide.element
import com.github.qky666.selenidepom.pom.Required
import com.github.qky666.selenidepom.test.kotlin.mtp.pom.common.MainFramePage

class ServicesRequiredErrorPage : MainFramePage() {
    @Required val principal = element("div.servicios-principal")
    @Required val title = principal.find("h1.h2")
    // badSelector generates an error in shouldLoadRequired
    @Required val badSelector = element("bad-selector")
    @Required val otherBadSelector = element("other-bad-selector")
}

val servicesRequiredErrorPage = ServicesRequiredErrorPage()
